File: resumedetective/paths.py
"""路径常量：所有数据统一存储在 data/ 目录下"""
import sys
from pathlib import Path

# 项目根目录
#  - 源码模式：__file__ 所在目录（ResumeDetective/）
#  - PyInstaller exe 模式：exe 所在目录（避免写入只读的 _internal/）
if getattr(sys, 'frozen', False):
    ROOT_DIR = Path(sys.executable).parent
else:
    ROOT_DIR = Path(__file__).parent.parent  # 包路径 → 项目根目录

# 数据目录
DATA_DIR = ROOT_DIR / "data"

# 配置文件（普通配置，不含敏感信息）
CONFIG_FILE = DATA_DIR / "config.json"

# 数据库文件
DB_FILE = DATA_DIR / "data.db"

# 简历文件存储目录
RESUMES_DIR = DATA_DIR / "Resumes"

# 附件目录（按 application_id 分目录）
ATTACHMENTS_DIR = DATA_DIR / "Attachments"

# Reasonix 本地数据目录（配置、日志等）
REASONIX_DATA_DIR = DATA_DIR / "reasonix"

# Skill 目录（可选，用于 Reasonix CLI 扩展）
SKILL_DIR = ROOT_DIR / "skills"

# Reasonix CLI 执行文件目录（应用内可控副本）
REASONIX_CLI_DIR = ROOT_DIR / "Reasonix Cli"
REASONIX_CLI_EXE = REASONIX_CLI_DIR / "reasonix.exe"

File: internal/config/paths.go
package config

import (
	"errors"
	"os"
	"path/filepath"
	"strings"
)

const DataDirEnv = "RESUME_DETECTIVE_V4_DATA_DIR"

type Paths struct {
	DataDir        string
	Database       string
	Workbook       string
	ConfigFile     string
	EnvFile        string
	SecretFile     string
	MigrationFile  string
	ResumesDir     string
	AttachmentsDir string
	BackupsDir     string
	UpdatesDir     string
	ReasonixDir    string
}

func Resolve(dataDir string) (Paths, error) {
	if dataDir == "" {
		dataDir = os.Getenv(DataDirEnv)
	}
	if dataDir == "" {
		executable, err := os.Executable()
		if err != nil {
			return Paths{}, err
		}
		dataDir = filepath.Join(filepath.Dir(executable), "data")
	}
	abs, err := filepath.Abs(dataDir)
	if err != nil {
		return Paths{}, err
	}
	if filepath.Clean(abs) == filepath.VolumeName(abs)+string(filepath.Separator) {
		return Paths{}, errors.New("数据目录不能是磁盘根目录")
	}
	backupsDir := filepath.Join(abs, "backups")
	if strings.EqualFold(filepath.Base(abs), "data") {
		backupsDir = filepath.Join(filepath.Dir(abs), "backups")
	}
	p := Paths{
		DataDir:        abs,
		Database:       filepath.Join(abs, "resume_detective.db"),
		Workbook:       filepath.Join(abs, "秋招投递追踪.xlsx"),
		ConfigFile:     filepath.Join(abs, "config.json"),
		EnvFile:        filepath.Join(abs, ".env"),
		SecretFile:     filepath.Join(abs, "secret.json.enc"),
		MigrationFile:  filepath.Join(abs, "v3-migration.json"),
		ResumesDir:     filepath.Join(abs, "resumes"),
		AttachmentsDir: filepath.Join(abs, "attachments"),
		BackupsDir:     backupsDir,
		UpdatesDir:     filepath.Join(abs, "updates"),
		ReasonixDir:    filepath.Join(abs, "reasonix"),
	}
	for _, dir := range []string{p.DataDir, p.ResumesDir, p.AttachmentsDir, p.BackupsDir, p.UpdatesDir, p.ReasonixDir} {
		if err := os.MkdirAll(dir, 0o700); err != nil {
			return Paths{}, err
		}
	}
	probe := filepath.Join(p.DataDir, ".write-test")
	if err := os.WriteFile(probe, []byte("ok"), 0o600); err != nil {
		return Paths{}, errors.New("程序所在目录不可写；请将便携版解压到个人文件夹后再运行")
	}
	_ = os.Remove(probe)
	return p, nil
}
